### PR TITLE
System: Refactor CPU Core management and move ARMInterface and Schedulers to Kernel

### PR DESCRIPTION
**Pull request by FernandoS27**
This PR aims to refactor our CPU core handling and at the same time remove some global instances of the system within the kernel.

### CODE REVIEW

File: src/core/cpu_manager.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+#include "core/gdbstub/gdbstub.h"
+#include "core/settings.h"
+
+namespace Core {
+
+CpuManager::CpuManager(System& system) : system{system} {}
+CpuManager::~CpuManager() = default;
+
+void CpuManager::Initialize() {
+
```
Comment (FearlessTobi) [inline src/core/cpu_manager.cpp]: why the newline?

File: src/core/core_manager.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 yuzu emulator team
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <atomic>
+#include <condition_variable>
```
Comment (lioncash) [inline src/core/core_manager.h]: `<condition_variable>` and `<mutex>` aren't used anywhere in this header.

File: src/core/arm/exclusive_monitor.h [hunk 3]
```diff
@@ -1,11 +1,17 @@
-// Copyright 2018 yuzu emulator team
+// Copyright 2020 yuzu emulator team
```
Comment (lioncash) [inline src/core/arm/exclusive_monitor.h]: This doesn't need to be changed. This is only to establish a creation year. Not the year this was last updated.

File: src/core/hle/kernel/physical_core.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+class ExclusiveMonitor;
+class System;
+} // namespace Core
+
+namespace Kernel {
+
+class PhysicalCore {
+public:
+    PhysicalCore(Core::System& system, KernelCore& kernel, std::size_t id, Core::ExclusiveMonitor& exclusive_monitor);
+
```
[the remark this answers is not in the document]
Comment (lioncash) [reply src/core/hle/kernel/physical_core.h]: This still applies

File: src/core/hle/kernel/physical_core.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
```
[the remark this answers is not in the document]
Comment (lioncash) [reply src/core/hle/kernel/physical_core.h]: This still applies

File: src/core/hle/kernel/physical_core.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (FernandoS27) [reply src/core/hle/kernel/physical_core.h]: getting issues by setting it up.

[the remark this answers is not in the document]
Comment (lioncash) [reply src/core/hle/kernel/physical_core.h]: What do you mean?

[the remark this answers is not in the document]
Comment (FernandoS27) [reply src/core/hle/kernel/physical_core.h]: It has been fixed now.

**Pull request merged**